Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -139,191 +138,175 @@ class Intersection {
   /// The Status of the intersection
   IntersectionStatus m_status = IntersectionStatus::unreachable;
 
-  constexpr Intersection() = default;
+  constexpr Intersection() noexcept = default;
 };
 
 using Intersection2D = Intersection<2>;
 using Intersection3D = Intersection<3>;
 
-static_assert(std::is_trivially_move_constructible_v<Intersection2D>);
 static_assert(std::is_trivially_copy_constructible_v<Intersection2D>);
+static_assert(std::is_trivially_move_constructible_v<Intersection2D>);
 static_assert(std::is_trivially_move_assignable_v<Intersection2D>);
 
-static constexpr std::uint8_t s_maximumNumberOfIntersections = 2;
-using MultiIntersection3D =
-    boost::container::static_vector<Intersection3D,
-                                    s_maximumNumberOfIntersections>;
+using IntersectionIndex = std::uint8_t;
+static constexpr IntersectionIndex s_maximumNumberOfIntersections = 2;
 
-class SurfaceIntersection {
+template <unsigned int DIM>
+class MultiIntersection {
```

Comment:
So this makes `MultiIntersection` directly iterable, to allow iterating over the `array` (vs. static vector before).
The fact that the iterator deref returns a pair including the index is a bit odd to me: is that not always the *naive* index? In that case, why not just use `Acts::enumerate` instead of implementing this specifically in this class?

Reply:
I think the only difference is the index type and the pair order. I will try to make it work with `Acts::enumerate`